### PR TITLE
[oxigraph_server] Add Oxigraph RDF DB

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

**Comment by jeremiahpslewis** [comment]
@giordano Unbelievable how much better the Yggdrasil rust binary experience has become over the past month or so, thank you!!!

File: O/oxigraph/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+sources = [
+    GitSource("https://github.com/oxigraph/oxigraph.git", "a21dcbb4f7355d7a00a86fbc5ad2c350a53629c4"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/oxigraph/lib
+
+cargo build --release -j${nproc}
```
Comment (giordano) [inline O/oxigraph/build_tarballs.jl:18]:
```suggestion
cargo build --release
```

File: O/oxigraph/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+install_license $WORKSPACE/srcdir/oxigraph/LICENSE-MIT
+"""
```
Comment (giordano) [inline O/oxigraph/build_tarballs.jl:21]: Probably need to install also the product?

File: O/oxigraph/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(; experimental=true)
+
+# Rust toolchain for i686 Windows is unusable
+filter!(p -> !Sys.iswindows(p) || arch(p) != "i686", platforms)
+
+# The products that we will ensure are always built
+products = Product[
```
Comment (giordano) [inline O/oxigraph/build_tarballs.jl:31]: Remember to add the products

File: O/oxigraph/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "oxigraph"
+version = v"0.2.5"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/oxigraph/oxigraph.git", "a21dcbb4f7355d7a00a86fbc5ad2c350a53629c4"),
+    DirectorySource("./bundled"),
```
Comment (giordano) [inline O/oxigraph/build_tarballs.jl:11]: Am I missing something or the files in this directory aren't actually used?

**Comment by jeremiahpslewis** [comment]
Build failures seem to have something to do with OpenSSL...but can't pin it down

**Comment by giordano** [comment]
That's because of https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/188, but that should have been worked around by https://github.com/JuliaPackaging/Yggdrasil/pull/4069

**Comment by jeremiahpslewis** [comment]
Hmm...glad it's not in my end. ;). Should I deactivate the platforms or wait? Don't really care about the affected platforms tbh, but no rush on this either

**Comment by jeremiahpslewis** [comment]
One error message that pops up after trying various linking strategies is:

```bash
  = note: /opt/i686-linux-musl/bin/../lib/gcc/i686-linux-musl/7.1.0/../../../../i686-linux-musl/bin/ld: warning: libc.musl-i386.so.1, needed by /workspace/destdir/lib/libssl.so, not found (try using -rpath or -rpath-link)
```

Feels like OpenSSL needs to be rebuilt / relinked with libc?

File: O/oxigraph_server/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/oxigraph/oxigraph.git", "a21dcbb4f7355d7a00a86fbc5ad2c350a53629c4"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/oxigraph/server
+
+export OPENSSL_STATIC=yes
```
Comment (giordano) [inline O/oxigraph_server/build_tarballs.jl:17]: Why static?  If that's static, OpenSSL is probably not a runtime dependency then?

Comment (jeremiahpslewis) [reply O/oxigraph_server/build_tarballs.jl:17]: It doesn't build if not static, so I'll change it to a build dependency?

Comment (giordano) [reply O/oxigraph_server/build_tarballs.jl:17]: Actually the program does link to OpenSSL:
```julia
shell> $(libtree()) oxigraph_server
oxigraph_server 
├── libssl.so.1.1 [default path]
│   └── libcrypto.so.1.1 [default path]
├── librt.so.1 [default path]
└── libcrypto.so.1.1 [default path]

julia> using libtree_jll

shell> $(libtree()) oxigraph_server
oxigraph_server 
├── libssl.so.1.1 [default path]
│   └── libcrypto.so.1.1 [default path]
├── librt.so.1 [default path]
└── libcrypto.so.1.1 [default path]
```
So I'm not sure what `OPENSSL_STATIC=yes` means.

File: O/oxigraph_server/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+filter!(p -> !Sys.iswindows(p) || arch(p) != "i686", platforms)
+
+# The products that we will ensure are always built
+products = Product[
+    ExecutableProduct("oxigraph_server", :oxigraph_server),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    BuildDependency("OpenSSL_jll"),
```
Comment (giordano) [inline O/oxigraph_server/build_tarballs.jl:40]:
```suggestion
    Dependency("OpenSSL_jll"),
```

File: O/oxigraph_server/build_tarballs.jl (hunk 5 above)
Comment (giordano) [reply O/oxigraph_server/build_tarballs.jl:17]: Where did you find [`OPENSSL_STATIC`](https://github.com/oxigraph/oxigraph/search?q=OPENSSL_STATIC)?

Comment (jeremiahpslewis) [reply O/oxigraph_server/build_tarballs.jl:17]: https://github.com/JuliaPackaging/Yggdrasil/blob/4dc836674457ca90a72830e4d4a4f80aa27278db/0_RootFS/Rust/build_tarballs.jl#L65

Comment (giordano) [reply O/oxigraph_server/build_tarballs.jl:17]: That doesn't explain what's the purpose of that line

Comment (jeremiahpslewis) [reply O/oxigraph_server/build_tarballs.jl:17]: Fair point.

Comment (jeremiahpslewis) [reply O/oxigraph_server/build_tarballs.jl]: Turns out, it is utterly unnecessary.

**Pull request merged**